simplq/src/components/common/Nav/LeftNav.jsx

```diff
@@ -1,28 +1,41 @@
 /* eslint-disable jsx-a11y/anchor-is-valid */
 
 import React from 'react';
-import { smoothScrollTo } from 'utils/scrollingOperations';
+import { smoothScrollTo, smoothScrollToHomePageTop } from 'utils/scrollingOperations';
+import { useHistory } from 'react-router';
 import styles from './Nav.module.scss';
 import LoginButton from '../LoginButton';
 
 const LeftNav = ({ open, toggleClose }) => {
+  const history = useHistory();
+
   const scrollToHowItWorks = () => {
     // Close the navbar on click
     toggleClose();
     // get the target div by ID
-    const element = document.getElementById('target_how_it_works');
+    let element = document.getElementById('target_how_it_works');
     if (element) {
       // element is on the current page, just have to scroll to it
       smoothScrollTo(element);
     } else {
-      window.location.href = '/#target_how_it_works';
+      history.push('/');
+      // wait till page loads before getting element
+      setTimeout(() => {
+        element = document.getElementById('target_how_it_works');
+        smoothScrollTo(element);
+      }, 0);
```

Comment:
lets create a util function onLoadbyId:
```
/** 
 * Execute a callback as soon as an element is available on the DOM.
 *
 * id - element id to wait on
 * callback - callback to execute as soon as the element becomes available. The 
 * element is passed to the callback and it is triggered.
function onLoadById(id, callback) {
  var checkAndExecute = setInterval(function() {
     const element = document.getElementById(id);
     if (element) {
        callback(element);
        clearInterval(checkAndExecute);
     }
  }, 100);
}

```

then just use it here: `onLoadById('target_how_it_works', e =>smoothScrollTo(e))`

Reply:
The above was inspired by [this](https://stackoverflow.com/a/16149679), you can check if there are better ways to do this, though.